Automattic/jetpack
# Social: Standardize the connection object in the recently added social endpoints

The newly added Jetpack Social endpoints don't return data in the same format as the initial state. This PR standardizes the response for the connection object.  I've also updated the `jetpack/v4/publicize/connections` endpoint which isn't used anywhere. 
We are getting rid of the site option and are going to rely on transient to cache the connections on the Jetpack site. 

## Proposed changes:

* Updated the create and update publicize connections endpoint to call the newly added `get_connections_for_user` publicize function. 
* Updated the get publicize connections endpoint to call the existing `get_all_connections_for_user` publicize function. 
* Migrated the `publicize_connections` to a transient to avoid issues with caching of the option. 
* Added `shared` key to the connections object. 
* Added `status` to the connections object. 
* To refresh connection test results, add this query param `run_test_results=true` to your get request. 
* To clear connection cache, add this query param `clear_cache =1` to your get request. 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Don't sandbox and test against prod. 
* Boot up a JN site on this branch or checkout this branch on your local and ensure `define( 'JETPACK_SOCIAL_USE_ADMIN_UI_V1', true ); 
`  is enabled or add the blog sticker to enable the feature. 
* Simulate the post, put and get requests to the following endpoints: `POST wp-json/jetpack/v4/social/connections`, PUT `wp-json/jetpack/v4/social/connections/:id` and GET `wp-json/jetpack/v4/publicize/connections` and observe that the API response.  

For the GET request, this will be the response.

```
{
    "refresh_token": "",
    "type": "access",
    "user_id": "",
    "provider": "nextdoor:wordpressxxxx",
    "issued": "2023-10-25 05:29:34",
    "expires": 1747224005,
    "external_id": "",
    "external_name": "Sid Darthan",
    "external_display": "Sid Darthan",
    "connection_data": {
        "id": "",
        "token_id": "",
        "blog_id": "",
        "user_id": "",
        "meta": {
            "external_user_id": ""
        }
    },
    "profile_picture": "xxxx",
    "service_name": "nextdoor",
    "connection_id": "",
    "can_disconnect": true,
    "profile_link": "https://nextdoor.com/profile/",
    "shared": false,
    "status": "ok"
}
``` 

 To get connection test results, include these query params `wp-json/jetpack/v4/publicize/connections?run_test_results=true`
When you create a connection by calling `https://temporary-lemming.jurassic.ninja/wp-json/jetpack/v4/social/connections` with the body `{"keyring_connection_ID" : xxx,  "shared" : false }`, you will see a similar response. Likewise for globalizing/unglobalizing as well.
